### ISSUE TITLE
Release 8.11 breaks compatibility with Python 3.5

### ISSUE THREAD

**Issue by mattbonnell** [more-itertools/more-itertools#578]
The new version makes use of f-strings, which were introduced in Python 3.6, making it incompatible with Python 3.5
```
File "<redacted>/.venv/lib/python3.5/site-packages/more_itertools/more.py", line 646

    f'Too few items in iterable (got {item_count})',

                                                  ^

SyntaxError: invalid syntax
```

**Comment by vfazio** [comment]
Note that 3.5 was dropped from the classifiers last October:
https://github.com/more-itertools/more-itertools/commit/c1bf0524e3e9cd73bdff7eba12e3cb50ab5801ed

And was dropped as part of 8.6 https://github.com/more-itertools/more-itertools/blob/master/docs/versions.rst#860

**Comment by mattbonnell** [comment]
The [`python-requires` param in the setup.py](https://github.com/more-itertools/more-itertools/blob/1ede288d19c2d47ddae8a14a2dc8f314cb925f90/setup.py#L38) needs to be bumped to prevent 3.5 users from automatically pulling the new version which they are no longer compatible with 

[Docs](https://packaging.python.org/guides/dropping-older-python-versions/#dropping-support-for-older-python-versions)

**Comment by andy-maier** [comment]
@mattbonnell Right. The python-requires param is what pip evaluates. The classifiers are not evaluated by pip. 

Circumvention for us was to exclude more-itertools 8.11.0 on Python 3.5.

**Comment by vfazio** [comment]
Right, but since it wasn't actually dropped from support, the only thing that could be referenced for the intention to no longer support it was the documentation changes.

Regardless, I've clarified my comment.

**Comment by mattbonnell** [comment]
Sorry, I don't quite follow what your point is.

**Comment by tysonite** [comment]
Can you remove release 8.11 on pypi? Or replace it with fixed version once #579 merged?

**Comment by webknjaz** [comment]
Oh, I was about to report the same issue. The runtime problem has been introduced in https://github.com/more-itertools/more-itertools/commit/ba5240aa7864eaf1d4e7ecd15818aea66ba1925c#diff-d7dff201bd69500a573b7e96c52f8e625242d769a04621224112faf6e7bc63fdR600.

Instead of removing, I think it's okay if the maintainers use `Yank`, I think. These are the versions that need to be yanked: v8.11.2, v8.11.1, v8.11.0.

**Comment by webknjaz** [comment]
Cross-posting other incarnations of the same bug that has happened in the past: https://github.com/more-itertools/more-itertools/issues/272 + https://github.com/more-itertools/more-itertools/issues/355.

**Comment by bbayles** [comment]
Bad news: I can't remove the old PyPI release.

Better news, I hope: I'll drop the f-strings and ship a new version.

**Comment by bbayles** [comment]
8.12.0 is now on PyPI with the f-strings dropped and 3.5 back in `setup.py`. Hopefully this fixes everybody's CI systems.

Somehow my GitHub notifications for non-mentions got turned off for this repository; apologies for not seeing this until today.

**Comment by hwalinga** [comment]
>Bad news: I can't remove the old PyPI release.

You should not remove it, but yank it instead: 

https://www.python.org/dev/peps/pep-0592/

**Comment by bbayles** [comment]
I don't have the permissions on PyPI, I'm afraid.

**Comment by bbayles** [comment]
@erikrose, if you're available, could you yank 8.11.0 on PyPI?

.1 and .2 were GitHub-only releases for an unrelated RTD.

**Comment by erikrose** [comment]
Yanked! Happy Yanksgiving!
